share/ci/scripts/linux/validate_openexr_libs.sh

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env bash
+
+#
+# Validate the libary symlinks:
+#   * The actual elf binary is, e.g. libIlmThread-3_1.so.29.0.0
+#   * The symlinks are:
+#       libIlmThread.so -> libIlmThread-3_1.so
+#       libIlmThread-3_1.so -> libIlmThread-3_1.so.29
+#       libIlmThread-3_1.so.29 -> libIlmThread-3_1.so.29.0.0
+#
+# Extract the version by compiling a program that prints the
+# OPENEXR_VERSION_STRING. This also validates that the program
+# compiles and executes with the info from pkg-config.
+# 
+
+if [[ $# == "0" ]]; then
+    echo "usage: $0 BUILD_ROOT [SRC_ROOT]"
+    exit -1
+fi
+
+BUILD_ROOT=$1
+SRC_ROOT=$2
+
+# Locate OpenEXR.pc and set PKG_CONFIG_PATH accordingly
+
+pkgconfig=$(find $BUILD_ROOT -name OpenEXR.pc)
+if [[ "$pkgconfig" == "" ]]; then
+    echo "Can't find OpenEXR.pc"
+    exit -1
+fi    
+export PKG_CONFIG_PATH=$(dirname $pkgconfig)
+
+# Build the validation program
+
+CXX_FLAGS=$(pkg-config OpenEXR --cflags)
+LD_FLAGS=$(pkg-config OpenEXR --libs --static)
+
+VALIDATE_CPP=$(mktemp --tmpdir "validate_XXX.cpp")
+VALIDATE_BIN=$(mktemp --tmpdir "validate_XXX")
+trap "rm -rf $VALIDATE_CPP $VALIDATE_BIN" exit
+
+echo -e '#include <ImfHeader.h>\n#include <OpenEXRConfig.h>\n#include <stdio.h>\nint main() { puts(OPENEXR_PACKAGE_STRING); Imf::Header h; return 0; }' > $VALIDATE_CPP
+
+g++ $CXX_FLAGS $VALIDATE_CPP -o $VALIDATE_BIN $LD_FLAGS
```

Comment:
will this work on the build matrix variants where we are using clang? (not sure if docker will put a gcc in there)

Reply:
It probably does make sense to use the same compiler as the build matrix, I'll look into a way to pass that in, although it works for now.

We recently had problems with installing broken symlinks and pkg-configs, so this is just a check in the ci to ensure that bug doesn't creep back in, not intended as a general utility or part of the test suite.